src/test/java/gov/va/viers/cdi/emis/client/MilitaryInfoClient.java

```diff
@@ -29,10 +38,41 @@
         input.setEdipiORicn(edi);
         JAXBElement<InputEdiPiOrIcn> request = factory.createEMISmilitaryServiceEligibilityRequest(input);
 
-        JAXBElement<EMISmilitaryServiceEligibilityResponseType> response =
-                (JAXBElement<EMISmilitaryServiceEligibilityResponseType>) webServiceTemplate.marshalSendAndReceive(request);
+    JAXBElement<EMISmilitaryServiceEligibilityResponseType> response =
+        (JAXBElement<EMISmilitaryServiceEligibilityResponseType>)
+            webServiceTemplate.marshalSendAndReceive(
+                request,
+                new WebServiceMessageCallback() {
 
+                  @Override
+                  public void doWithMessage(WebServiceMessage message) {
+                    try {
+                      // get the header from the SOAP message
+                      SoapHeader soapHeader = ((SoapMessage) message).getSoapHeader();
 
+                      // create the header element
+                      gov.va.viers.cdi.cdi.commonservice.v2.ObjectFactory objectFactory =
+                          new gov.va.viers.cdi.cdi.commonservice.v2.ObjectFactory();
+
+                      InputHeaderInfo header = objectFactory.createInputHeaderInfo();
+                      header.setSourceSystemName("sourceSystem");
+                      header.setUserId("userId");
+                      header.setTransactionId("transactionId");
+
+                      JAXBElement<InputHeaderInfo> headers =
```

Comment:
Give headers a different name since it's hard to distinguish from header. I might suggest jaxbHeader or something to show it's been wrapped